azure/services/async/async.go

```diff
@@ -183,12 +194,27 @@ func (s *Service) DeleteResource(ctx context.Context, spec azure.ResourceSpecGet
 	return nil
 }
 
-// retryAfter returns the max between the `RETRY-AFTER` header and the default requeue time.
+// getRequeueAfterFromFuture returns the max between the `RETRY-AFTER` header and the default requeue time.
 // This ensures we respect the retry-after header if it is set and avoid retrying too often during an API throttling event.
-func retryAfter(sdkFuture azureautorest.FutureAPI) time.Duration {
+func getRequeueAfterFromFuture(sdkFuture azureautorest.FutureAPI) time.Duration {
 	retryAfter, _ := sdkFuture.GetPollingDelay()
 	if retryAfter < reconciler.DefaultReconcilerRequeue {
 		retryAfter = reconciler.DefaultReconcilerRequeue
 	}
 	return retryAfter
 }
+
+// getRetryAfterFromDetailedError returns the time.Duration from the http.Response in the autorest.DetailedError
+// If there is no Response object, or if there is no meaningful Retry-After header data, we return a zero value time.Duration
+func getRetryAfterFromDetailedError(detailedError autorest.DetailedError) (ret time.Duration) {
+	if detailedError.Response != nil {
+		if retryAfter := detailedError.Response.Header.Get("Retry-After"); retryAfter != "" {
+			if rai, _ := strconv.Atoi(retryAfter); rai > 0 {
+				ret = time.Duration(rai) * time.Second
+			}
+		} else if detailedError.Response.StatusCode == http.StatusTooManyRequests {
```

Comment:
do we expect that `StatusTooManyRequests` error and `Retry-After` header being set are 2 mutually exclusive events?

Reply:
Makes sense. Could we add code comments? It wasn't completely clear to me without reading your explanation.

Reply:
Good point, done!